### DISCUSSION TITLE
Template adds additional element with x-if

### DISCUSSION THREAD

**Discussion by xorock** [alpinejs/alpine#4285] [7. Bugs (archived)]
Hi.

```html
<div
      x-data="{
        open: true
      }"
    >
      <template x-if="open" x-for="i in 2">
        <section>aaa</section>
      </template>
    </div>
```

Result
```
aaa
aaa
aaa
```

Expected result (valid when x-if="open" is removed)
```
aaa
aaa
```

If the element contains x-if="open" then the elements are repeated 3 times, some empty element without access to e.g. the index is added. When this is removed, the element duplicates correctly 2 times.

**Comment by ekwoka** [comment]
x-if and x-for are separate things.

They are both cloning the template, so of course it will do 3.

**Comment by xorock** [comment]
Ok, so I have to move condition directly to x-for, `<template x-for="i in open ? 2: 0">`.
I must admit that this is a surprise and rather unexpected in templates. But thank You for the clarification.

**Reply by ekwoka** [reply]
that's the best way for sure.
